numba/_listobject.c

```diff
@@ -384,6 +385,104 @@ numba_list_resize(NB_List *lp, Py_ssize_t newsize) {
     return LIST_OK;
 }
 
+/* Delete a slice 
+ *
+ * start: the start index of ths slice
+ * stop: the stop index of the slice (not included)
+ * step: the step to take
+ *
+ * This function assumes that the start and stop were clipped appropriately.
+ * I.e. if step > 0 start >= 0 and stop <= len(l) and
+ *      if step < 0 start <= length and stop >= -1
+ *      step != 0 and no Python negative indexing allowed.
+ *
+ * This code was copied and edited from the relevant section in
+ * list_ass_subscript from the cpython implementation.
```

Comment:
Please add exact URL with SHA pointing to source, thanks.

Reply:
The exact sources are listed at the top of the file, in the interest of deduplication, I added a comment pointing the reader to the top of the file.